xbmc/music/MusicDatabase.cpp

```diff
@@ -2180,6 +2223,101 @@ bool CMusicDatabase::ClearArtistLastScrapedTime(int idArtist)
   return ExecuteQuery(strSQL);
 }
 
+bool CMusicDatabase::AddArtistVideoLinks(const CArtist& artist)
+{
+  bool res = false;
+  std::string dbSong;
+  auto start = std::chrono::steady_clock::now();
+
+  try
+  {
+    if (nullptr == m_pDB || nullptr == m_pDS || nullptr == m_pDS2)
+      return res;
+
+    std::string strSQL = PrepareSQL(
+        "SELECT idSong, strTitle, strMusicBrainzTrackID FROM song WHERE idAlbum IN (SELECT "
+        "idAlbum FROM album_artist WHERE idArtist = %i)",
+        artist.idArtist);
+    if (!m_pDS->query(strSQL))
+      return res;
+    auto queryEnd = std::chrono::steady_clock::now();
+    auto queryDuration = std::chrono::duration_cast<std::chrono::milliseconds>(queryEnd - start);
+    while (!m_pDS->eof())
+    {
+      dbSong = m_pDS->fv(1).get_asString();
+      const std::string mbTrackID = m_pDS->fv(2).get_asString();
+      const int songId = m_pDS->fv(0).get_asInt();
+      for (const auto& videoURL : artist.videolinks)
+      {
+        if (mbTrackID == videoURL.mbTrackID ||
+            StringUtils::CompareNoCase(dbSong, videoURL.title.c_str()) == 0)
+        {
+          std::string strSQL2 = PrepareSQL("UPDATE song set strVideoURL='%s' WHERE idSong = %i",
+                                           videoURL.videoURL.c_str(), songId);
+          res = m_pDS2->exec(strSQL2);
+
+          if (!videoURL.thumbURL.empty())
+          {
+            // already have a thumb for this videoURL ?
```

Comment:
Existing art could be for a different url, so I would replace comment
`// Already have a videothumb for this song?`